### PR TITLE
Upgrade graphql-ruby into its latest minor version

### PR DESCRIPTION
**Pull request by skateman**
https://issues.redhat.com/browse/RHICOMPL-3036

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/graphql/schema.rb [hunk 1]
```diff
@@ -15,8 +15,8 @@ class Schema < GraphQL::Schema
   mutation Types::Mutation
   lazy_resolve(Promise, :sync)
   use GraphQL::Batch
-  use GraphQL::Execution::Interpreter
-  use GraphQL::Analysis::AST
+  # For batch-loading (see https://graphql-ruby.org/dataloader/overview.html)
+  # use GraphQL::Dataloader
```
Comment (romanblanco) [inline app/graphql/schema.rb:19]: Do we plan to enable this in a soon future?

File: app/graphql/mutations/profile/create.rb [hunk 2]
```diff
@@ -12,7 +12,6 @@ class Create < BaseMutation
       argument :name, String, required: true
       argument :ref_id, ID, required: true
       argument :description, String, required: false
-      argument :business_objective_id, String, required: false
```
Comment (vkrizan) [inline app/graphql/mutations/profile/create.rb:15]: Doublecheck whether FE does not use this.

File: app/graphql/types/query.rb [hunk 3]
```diff
@@ -13,7 +13,6 @@ class Query < Types::BaseObject
     collection_field :test_results, Types::TestResult
     collection_field :benchmarks, Types::Benchmark
     collection_field :os_major_versions, Types::OsMajorVersion
-    record_field :profile, Types::Profile
```
Comment (vkrizan) [inline app/graphql/types/query.rb:16]: Why is profile being removed?

File: app/graphql/schema.rb (hunk 1 above)
Comment (skateman) [reply app/graphql/schema.rb:19]: Yes!

File: app/graphql/mutations/profile/create.rb (hunk 2 above)
Comment (skateman) [reply app/graphql/mutations/profile/create.rb:15]: It's a duplicated argument, it's also defined as an ID and the newer version is stricter.

File: app/graphql/types/query.rb (hunk 3 above)
Comment (skateman) [reply app/graphql/types/query.rb:16]: Same here, duplication

**Pull request merged**